presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -186,6 +186,11 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         TableHandle newTable;
         TupleDomain<ColumnHandle> remainingFilter;
         if (!metadata.usesLegacyTableLayouts(session, node.getTable())) {
+            if (!constraint.predicate().isPresent() && newDomain.contains(node.getEnforcedConstraint())) {
+                // new domain is wider than domain already provided by table scan
+                return Optional.empty();
```

Comment:
Shall we eliminate Filter in such a case?

(Consider adding a TODO for now, unless the answer is obvious.)